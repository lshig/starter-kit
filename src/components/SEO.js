import React from 'react';
import { Helmet } from 'react-helmet';

export default function SEO() {
  const seo = {
    article: null,
    author: 'Liz Shigetoshi',
    description: 'Template for building a React app from scratch',
    image: null,
    keywords: 'template,react,babel,webpack',
    title: 'starter-kit',
    twitterUsername: '@LizShigetoshi',
    url: 'https://lizshigetoshi.com/starter-kit'
  };

  return (
    <Helmet title={seo.title}>
      <meta name="author" content={seo.author} />
      <meta name="description" content={seo.description} />
      {(seo.image ? true : null) && <meta name="image" content={seo.image} />}
      <meta name="keywords" content={seo.keywords} />
      {seo.url && <meta property="og:url" content={seo.url} />}
      {(seo.article ? true : null) && (
        <meta property="og:type" content="article" />
      )}
      {seo.title && <meta property="og:title" content={seo.title} />}
      {seo.description && (
        <meta property="og:description" content={seo.description} />
      )}
      {(seo.image ? true : null) && (
        <meta property="og:image" content={seo.image} />
      )}
      <meta name="twitter:card" content="summary_large_image" />
      {seo.twitterUsername && (
        <meta name="twitter:creator" content={seo.twitterUsername} />
      )}
      {seo.title && <meta name="twitter:title" content={seo.title} />}
      {seo.description && (
        <meta name="twitter:description" content={seo.description} />
      )}
      {(seo.image ? true : null) && (
        <meta name="twitter:image" content={seo.image} />
      )}
    </Helmet>
  );
}
