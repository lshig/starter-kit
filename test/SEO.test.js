import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SEO from '../src/components/SEO';

describe('SEO', () => {
  describe('basic', () => {
    const wrapper = shallow(<SEO />);

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
