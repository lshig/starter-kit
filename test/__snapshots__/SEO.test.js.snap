// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO basic renders 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  title="starter-kit"
>
  <meta
    content="Liz Shigetoshi"
    name="author"
  />
  <meta
    content="Template for building a React app from scratch"
    name="description"
  />
  <meta
    content="template,react,babel,webpack"
    name="keywords"
  />
  <meta
    content="https://lizshigetoshi.com/starter-kit/"
    property="og:url"
  />
  <meta
    content="starter-kit"
    property="og:title"
  />
  <meta
    content="Template for building a React app from scratch"
    property="og:description"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@LizShigetoshi"
    name="twitter:creator"
  />
  <meta
    content="starter-kit"
    name="twitter:title"
  />
  <meta
    content="Template for building a React app from scratch"
    name="twitter:description"
  />
</HelmetWrapper>
`;
